ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -56,6 +57,10 @@
       Pattern.compile("\\s*UNSET\\s+'((?:[^']*|(?:''))*)'\\s*;\\s*");
   private static final Pattern DROP_CONNECTOR =
       Pattern.compile("\\s*DROP\\s+CONNECTOR\\s+(.*?)\\s*;\\s*");
+  private static final Pattern DEFINE_VARIABLE = Pattern.compile(
+      "\\s*DEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*=\\s*'((?:[^']*|(?:''))*)'\\s*;\\s*");
+  private static final Pattern UNDEFINE_VARIABLE =
+      Pattern.compile("\\s*UNDEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*;\\s*");
```

Comment:
Where did this set of valid characters in the variable name (i.e., `[a-zA-Z_][a-zA-Z0-9_@]*`) come from?

Reply:
The variable name follows the pattern for identifiers in the parser: https://github.com/confluentinc/ksql/blob/01bb840c0142d6e4a0c51b7988c0cb30b01f1a60/ksqldb-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4#L582

Reply:
OK. It's not great that we've got it in two places now (the parser and also the migrations tool) but there are disadvantages to having them be shared as well. How about we add a comment to this line in the migrations tool code to reference the SqlBase.g4 line (with a git permalink) so if the parser is changed in the future and something breaks, we'll at least know where to look to update this.